lib/opensearch-aws-sigv4.rb

```diff
@@ -37,7 +37,42 @@ module Aws
     #
     # @attr [Aws::Sigv4::Signer] sigv4_signer Signer used to sign every request
     class Sigv4Client < ::OpenSearch::Client
```

Comment:
It's shouldn't extend OpenSearch::Client anymore. The new implementation can now be independent of the `opensearch-ruby` gem. This gem is now much simpler: A class that accepts options to create an instance of aws-sigv4 and a logger object, and supports a `sign_request` method according to this [guide](https://github.com/opensearch-project/opensearch-ruby/blob/main/DEVELOPER_GUIDE.md#create-a-request-signer).

Reply:
Basically it's an entirely different implementation of sigv4 so we will need to bump the major version as well.